CeeFM/Decideify
# Random date generator for NYT Books API / Decideify Books Algorithm!

**Description**
This pull request updates the Books.js suggestion algorithm to automatically append a random date between 6/8/2008 and the current day to the NYT Books API GET request in the APIManager.js. This makes it so that every call to the NYT Books API will get a random date passed in, creating more variety in the summary of best sellers lists we get back in return.

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- You can go to the Books (/books) content hub and press the "Test Books API" button (nothing should happen in the console or browser, simply a fetch call is triggered and data is returned) followed by pressing the "Print Books" - which will print the full results of the NYT's Best Sellers Lists overviews for the week that is closest to the random date that was dynamically put into the fetch call.
- You can test this by opening the first console.log result that happens after your "Print Books" button click - the published date should change after every fetch call (which is the "Test Books API") and then can be confirmed with this process after clicking the "Print Books" button.

Checklist
